src/client/serialize/daos_serialize.c

```diff
@@ -1175,3 +1252,1807 @@ daos_cont_deserialize_attrs(char *filename, uint64_t *num_attrs,
+	if (status < 0) {
+		D_ERROR("Failed to write akey dset\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+out:
+	if (akey_dspace >= 0)
+		H5Sclose(akey_dspace);
+	if (akey_memtype >= 0)
+		H5Tclose(akey_memtype);
+	if (akey_vtype >= 0)
+		H5Tclose(akey_vtype);
+	if (akey_dset >= 0)
+		H5Dclose(akey_dset);
+	return rc;
+}
+
+static int
+serialize_version(struct hdf5_args *hdf5, float version)
+{
+	int	rc = 0;
+	hid_t	status = 0;
+	char	*version_name = "Version";
+	hsize_t	version_attr_dims[1];
+	hid_t	version_attr_type = -1;
+	hid_t	version_attr_dspace = -1;
+	hid_t	version_attr = -1;
+
+	version_attr_dims[0] = 1;
+	version_attr_type = H5Tcopy(H5T_NATIVE_FLOAT);
+	status = H5Tset_size(version_attr_type, 4);
+	if (status < 0) {
+		D_ERROR("Failed to set version attribute size\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	if (version_attr_type < 0) {
+		D_ERROR("Failed to create version attribute datatype\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	version_attr_dspace = H5Screate_simple(1, version_attr_dims, NULL);
+	if (version_attr_dspace < 0) {
+		D_ERROR("Failed to create version attribute dataspace\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	version_attr = H5Acreate2(hdf5->file, version_name, version_attr_type, version_attr_dspace,
+				  H5P_DEFAULT, H5P_DEFAULT);
+	if (version_attr < 0) {
+		D_ERROR("Failed to create version attribute\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	status = H5Awrite(version_attr, version_attr_type, &version);
+	if (status < 0) {
+		D_ERROR("Failed to write attribute\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+out:
+	if (version_attr_type >= 0)
+		H5Tclose(version_attr_type);
+	if (version_attr_dspace >= 0)
+		H5Sclose(version_attr_dspace);
+	if (version_attr >= 0)
+		H5Aclose(version_attr);
+	return rc;
+}
+
+int
+daos_cont_serialize(daos_prop_t *props, int num_attrs, char **names, char **buffers, size_t *sizes,
+		    int *total_oids, int *total_dkeys, int *total_akeys, uint64_t *bytes_read,
+		    daos_handle_t coh, char *filename)
+{
+	int			rc = 0;
+	int			rc2 = 0;
+	hid_t			status = 0;
+	int			i = 0;
+	int			dkey_index = 0;
+	int			akey_index = 0;
+	struct hdf5_args	hdf5 = {0};
+	daos_anchor_t		anchor;
+	daos_epoch_t		epoch;
+	daos_epoch_range_t	epr;
+	daos_handle_t		toh;
+	daos_obj_id_t		oids[OID_ARR_SIZE];
+	daos_handle_t		oh;
+	uint32_t		oids_nr;
+	bool			is_kv = false;
+	float			version = 0.0;
+	hid_t			usr_attr_memtype = -1;
+	hid_t			usr_attr_name_vtype = -1;
+	hid_t			usr_attr_val_vtype = -1;
+	struct oid		*oid_tmp = NULL;
+
+	if (filename == NULL)
+		D_GOTO(out, rc = -DER_INVAL);
+
+	/* Create HDF5 file for serialization */
+	hdf5.file = H5Fcreate(filename, H5F_ACC_TRUNC, H5P_DEFAULT, H5P_DEFAULT);
+	if (hdf5.file < 0) {
+		D_ERROR("Failed to create HDF5 file: %s\n", filename);
+		D_GOTO(out, rc = -DER_IO);
+	}
+
+	D_PRINT("Serializing Container to: %s\n", filename);
+
+	/* serialize cont version */
+	rc = serialize_version(&hdf5, version);
```

Comment:
Seems it's support to be the version of the serialization tool, for future compatibility checks:
https://github.com/daos-stack/daos/blob/efe058a0e61a2f978c12da3003cc29761ef43a9c/src/client/serialize/daos_serialize.c#L16
It can probably just be an integer